cumulus/client/consensus/aura/src/collators/mod.rs

```diff
@@ -173,6 +175,36 @@ async fn cores_scheduled_for_para(
 		.collect()
 }
 
+// Return all the cores assigned to the para at the provided relay parent, using the claim queue
+// offset. This assumes the relay chain runtime supports the claimqueue runtime API.
+// Will return an empty vec if the provided offset is higher than the claim queue length (which
+// corresponds to the scheduling_lookahead on the relay chain).
+async fn cores_scheduled_for_para(
```

Comment:
I think we can actually return the exact scheduled core we need here. The runtime gives us the cq offset and we can have another runtime api to give us the core selector. We could just use the block number but a runtime API is a better interface in case we change the underlying sequence number in the future.